### PR TITLE
N+1 problems

### PR DESCRIPTION
**Pull request by Kwaku28**
This part of the project is implemented to fix the n+1 problem

Tasks completed include the following:

1. Refactor code to fix the n+1 problem in the `See all posts` page
    - Before
       app-log(before)

    - After
      app-log(after)

2. Added Integration Specs to the project
3. Used Capybara to write the integration tests for each view in our project
     - Added tests for the `User` index page
     - Added tests for the `User` show page
     - Added tests for the `Post` index page
     - Added tests for the `Post` show page

Thank you🤝

### CODE REVIEW

File: spec/features/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    it 'should redirected to that user show page, When I click on a user' do
+      click_link 'Lebron'
+      expect(page).to have_current_path("/users/#{@user1.id}")
+    end
+  end
+
+  describe 'user#show' do
+    before(:each) do
+      @post1 = Post.create(title: 'post 1 title',
+                           text: 'post content 1',
```
Comment (thecodechaser) [inline spec/features/user_spec.rb]: - [ ] Please make sure to separate the tests for the users index and show actions in different files. Example: `user_index_spec.rb` and `user_show_spec.rb`.

File: spec/features/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      click_link 'post 1 title'
+      expect(page).to have_content 'post 1 title by Lawrence Amoafo'
+    end
+  end
+  describe 'post show' do
+    before(:each) do
+      visit "/users/#{@user1.id}/posts/#{@post1.id}"
+    end
+    it 'I can see the posts title.' do
+      expect(page).to have_content 'post 1 title'
```
Comment (thecodechaser) [inline spec/features/post_spec.rb]: - [ ] Please make sure to separate the tests for the posts index and show actions in different files. Example: `post_index_spec.rb` and `psot_show_spec.rb`.

File: spec/features/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      expect(page).to_not have_content 'Comments: 0'
+    end
+    it 'I can see how many likes a post has.' do
+      expect(page).to_not have_content 'Likes: 1'
+      expect(page).to have_content 'Likes: 0'
+    end
+    it 'When I click on a post, it redirects me to that posts show page.' do
+      click_link 'post 1 title'
+      expect(page).to have_content 'post 1 title by Lawrence Amoafo'
+    end
```
Comment (thecodechaser) [inline spec/features/post_spec.rb]: - [ ] You missed to add the below test for the posts index action, Please make sure to add that and it should pass without any errors.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Kwaku28, and @joyapisi👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
- [ ] Please make sure the N+1 problem is solved when fetching all posts and their comments for a user.
  - Example in the posts#index action `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts`

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_spec.rb (hunk 3 above)
Comment (Kwaku28) [reply spec/features/post_spec.rb]: Test added

**Review by Owusu-Desmond — APPROVED** [review]
Hi @Kwaku28 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**